src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -185,6 +190,24 @@ def make_requires_python_requirement(self, specifier):
             return None
         return RequiresPythonRequirement(specifier, self._python_candidate)
 
+    def get_wheel_cache_entry(self, link, name):
+        # type: (Link, Optional[str]) -> Optional[CacheEntry]
+        """Look up the link in the wheel cache.
+
+        If ``preparer.require_hashes`` is True, don't use the wheel cache,
+        because cached wheels, always built locally, have different hashes
+        than the files downloaded from the index server and thus throw false
+        hash mismatches. Furthermore, cached wheels at present have
+        undeterministic contents due to file modification times.
+        """
+        if self._wheel_cache is None or self.preparer.require_hashes:
+            return None
+        return self._wheel_cache.get_cache_entry(
+            link=link,
+            package_name=name,
+            supported_tags=get_supported(),
```

Comment:
Does this need to consider `--platform` and similar? (I really hope not, as it would be quite difficult to get at that data from here!)

Reply:
It does not; the cache is link-based, the platform filtering is done in the finder.

Reply:
Cool, this looks good to me then.